pkg/sql/virtual_table.go

```diff
@@ -24,20 +25,95 @@ import (
 // nil). If there is an error, then return (nil, error).
 type virtualTableGenerator func() (tree.Datums, error)
 
+type virtualTableGeneratorResponse struct {
+	datums tree.Datums
+	err    error
+}
+
+var errRowsDone = errors.New("predefined error that the generator has been canceled")
```

Comment:
i would remove the "predefined error" bit in cast the user gets this and gets confused

Reply:
Done